pkg/parsing/query_validator.go

```diff
@@ -9,16 +9,25 @@ import (
 	"github.com/textileio/go-tableland/pkg/tables"
 )
 
-// Stmt represents any valid read or mutating query.
-type Stmt interface {
-	GetQuery() (string, error)
+// WriteQueryResolver resolves Tablealand Custom Functions for a write statement.
+type WriteQueryResolver interface {
+	// GetTxnHash returns the transaction hash of the transaction containing the query being processed.
+	GetTxnHash() string
+
+	// GetBlockNumber returns the block number of the block containing query being processed.
+	GetBlockNumber() int64
+}
```

Comment:
I think the `WriteQueryResolver` and `ReadQueryResolver` should be defined in the `go-sqlparser` to avoid `go-sqlparser` depending on `go-tabeland`. 

That would change [L44](https://github.com/tablelandnetwork/go-tableland/pull/424/files#diff-c6920746c771dcedad8850e6ba75a3e2b487df3092caab3b6a8ee9dbb0cf5f50R44) to:
`GetQuery(sqlparser.WriteQueryResolver) (string, error)`